include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -356,7 +356,7 @@ template<typename T NBL_FUNC_REQUIRES(is_floating_point_v<T> && !is_matrix_v<T>)
 conditional_t<is_vector_v<T>, vector<bool, vector_traits<T>::Dimension>, bool> FOrdEqual(T lhs, T rhs);
 
 
-template<typename T, typename U NBL_FUNC_REQUIRES(!is_matrix_v<T> && !is_matrix_v<U> && is_same_v<typename vector_traits<U>::scalar_type, bool>)
+template<typename T, typename U NBL_FUNC_REQUIRES(!is_matrix_v<T> && (is_scalar_v<U> || (is_vector_v<U> && is_vector_v<T> && vector_traits<U>::Dimension==vector_traits<T>::Dimension)) && is_same_v<typename vector_traits<U>::scalar_type, bool>)
```

Comment:
`!is_matrix_v<T> && ` shouldn't be there as `select(BooleanScalar,Composite,Composite)` is allowed since version 1.4
https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpSelect

 and a matrix is a Composite apparently
 https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#CompositeType